wordpress-mobile/WordPress-iOS
# Site Creation: error view

Fixes #8555 

**To test:**

- Go through the site creation process.
- Trigger an error. 😄 
- @nate - let me know if want tips on triggering errors. There are two overall types: 
  - validation - created by a failure in `SiteCreationFields.validateFields`.
  - site create - created by a failure in any step in `SiteCreationService`.

- Verify the view looks like the 'implemented' image below.
- NOTE: The design has a 'Back' nav bar button. However, as discussed below, there is a 'Dismiss' button that exits the Site Creation process (i.e. dismisses the SC nav controller).
- The bold title will always be the same.
- The subtitle will change depending on the error triggered.

**Validation errors, messages and actions:**
NOTE: These are "sanity" validations. Since we do validation on each step prior to the site creation step, these will most likely never appear.
- site title: 
    - "The Site Title is missing."
    - 'Try again' goes to Site Details.
- domain: 
    - "The Site Domain is missing."
    - "The Site Domain contains wordpress.com."
    - 'Try again' goes to Domain Suggestion.
- theme: 
    - "The Site Theme is missing."
    - 'Try again' goes to Theme Selection.

**Site Creation errors, messages and actions:**
- validation: 
    - "The Site Domain is invalid."
    - 'Try again' goes to Domain Suggestion.
- getting default account: 
    - "We were unable to get your account information."
    - 'Try again' goes to Create Site, re-attempting site creation.
- creating site: 
    - "We were unable to create the site."
    - 'Try again' goes to Create Site, re-attempting site creation.
- setting tagline: 
    - "Your Site was created. Unfortunately, we were unable to set the Site Tagline. You can set the Tagline in the Site Settings, or try again now."
    - 'Try again' goes to Create Site, re-trying the process starting from setting the tagline.
- setting theme: 
    - "Your Site was created. Unfortunately, we were unable to set the Site Theme. You can set the Theme in the Site Settings, or try again now."
    - 'Try again' goes to Create Site, re-trying the process starting from setting the theme.
- syncing account: 
    - "Your Site was created. Unfortunately, we were unable to sync your account information. You can still access your site from My Sites. You can also retry syncing your account now."
    - 'Try again' goes to Create Site, re-trying the process starting from syncing the account.

**Implemented:**
![error_implemented](https://user-images.githubusercontent.com/1816888/35755870-3af3a172-0826-11e8-818f-b64ccfa584af.png)

**Designed:**
<img width="300" alt="error_designed" src="https://user-images.githubusercontent.com/1816888/35755873-4159355e-0826-11e8-8fb1-ac01b687c673.png">
